base_layer/wallet/src/utxo_scanner_service/utxo_scanning.rs

```diff
@@ -254,11 +255,20 @@ where TBackend: WalletBackend + 'static
             Ok(conn) => Ok(conn),
             Err(e) => {
                 self.publish_event(UtxoScannerEvent::ConnectionFailedToBaseNode {
-                    peer,
+                    peer: peer.clone(),
                     num_retries: self.num_retries,
                     retry_limit: self.retry_limit,
                     error: e.to_string(),
                 });
+                // No use re-dialing a peer that is not responsive for recovery mode
+                if self.mode == UtxoScannerMode::Recovery {
+                    if let Ok(Some(connection)) = self.resources.comms_connectivity.get_connection(peer.clone()).await {
+                        if connection.clone().disconnect().await.is_ok() {
+                            debug!(target: LOG_TARGET, "Disconnected base node peer {}", peer);
+                        }
+                    };
+                    sleep(Duration::from_secs(30));
```

Comment:
It is very bad to sleep a thread in an async task. If you really want to sleep you should use the tokio async sleep.